lib/gc/Transforms/DeepTileContractionOp.cpp

```diff
@@ -107,6 +107,16 @@ tensorViewRankedTensor(RewriterBase &rewriter, RankedTensorType outTensorType,
   return result;
 }
 
+// Check if the loop is dummy loop(has only one iteration)
+bool isDummyLoop(LoopLikeOpInterface loop) {
```

Comment:
Seems this is another impl besides "promoteIfSingleIteration()"?

Reply:
Yes, the purpose is similar but the `promoteIfSingleIteration()` erase the single iteration loop only after the loop is created. The extract/insert slices will be preserved, which causes the WAW conflict in bufferization and introduces more `memref.copy`. Here skip generating the single iteration loop and its corresponding extract/insert_slice op in the creation  phase once detecting the single iteration.

Reply:
I see, thanks for the explanation.